docs/src/docs/asciidoc/user/config-derive.adoc

```diff
@@ -0,0 +1,301 @@
+
+.adding a service creation configuration (constraining the default thread pool)
+[source,java,indent=0]
+----
+include::{sourcedir37}/impl/src/test/java/org/ehcache/docs/ConfigurationDerivation.java[tag=withServiceCreation]
+----
+[cols=".^~a,^.^~d,.^~a"]
+|===
+|[source,xml]
+----
+<ehcache>
+</ehcache>
+----
+|&rarr;
+|[source,xml]
+----
+<ehcache>
+  <thread-pools>
+    <thread-pool alias="default" default="true"
+                 min-size="1" max-size="16"/>
+  </thread-pools>
+</ehcache>
+----
+|===
+
+.updating a service creation configuration (changing the persistence path)
+[source,java,indent=0]
+----
+include::{sourcedir37}/impl/src/test/java/org/ehcache/docs/ConfigurationDerivation.java[tag=updateServiceCreation]
+----
+[cols=".^~a,^.^~d,.^~a"]
+|===
+|[source,xml]
+----
+<ehcache>
+  <persistence directory="/some/thing/else"/>
+</ehcache>
+----
+|&rarr;
+|[source,xml]
+----
+<ehcache>
+  <persistence directory="/var/persistence/path"/>
+</ehcache>
+----
+|===
+
+.adding a service configuration (setting a resilience strategy)
+[source,java,indent=0]
+----
+include::{sourcedir37}/impl/src/test/java/org/ehcache/docs/ConfigurationDerivation.java[tag=withService]
+----
+[cols=".^~a,^.^~d,.^~a"]
+|===
+|[source,xml]
+----
+<ehcache>
+  <cache alias="cache">
+    <key-type>Long.class</key-type>
+    <value-type>Object.class</value-type>
+    <heap>10</heap>
+  </cache>
+</ehcache>
+----
+|&rarr;
+|[source,xml]
+----
+<ehcache>
+  <cache alias="cache">
+    <key-type>Long.class</key-type>
+    <value-type>Object.class</value-type>
+    <heap>10</heap>
+    <resilience>
+      com.example.ThrowingResilienceStrategy
+    </resilience>
+  </cache>
+</ehcache>
+----
+|===
+
+.updating a service configuration (changing a clustered cache's consistency)
+[source,java,indent=0]
+----
+include::{sourcedir37}/clustered/client/src/test/java/org/ehcache/clustered/client/docs/ConfigurationDerivation.java[tag=updateService]
+----
+[cols=".^~a,^.^~d,.^~a"]
+|===
+|[source,xml]
+----
+<ehcache>
+  <service>
+    <tc:cluster>
+      <tc:connection url="terracotta://example.com/cachemanager"/>
+    </tc:cluster>
+  </service>
+
+  <cache alias="cache">
+    <resources>
+      <tc:clustered-dedicated unit="MB">50</tc:clustered-dedicated>
+    </resources>
+    <tc:clustered-store consistency="strong"/>
+  </cache>
+</ehcache>
+----
+|&rarr;
+|[source,xml]
+----
+<ehcache>
+  <service>
+    <tc:cluster>
+      <tc:connection url="terracotta://example.com/cachemanager"/>
+    </tc:cluster>
+  </service>
+
+  <cache alias="cache">
+    <resources>
+      <tc:clustered-dedicated unit="MB">50</tc:clustered-dedicated>
+    </resources>
+    <tc:clustered-store consistency="eventual"/>
+  </cache>
+</ehcache>
+----
+|===
+
+=== Removing a service
+Removing a service often involves removing both service creation and a service configuration instances since a service
+instance its configuration are usually strongly coupled:
+
+.removing a service
```

Comment:
This isn't just "removing a service" -- it's making a cache non-clustered!  This would have been a good one to show full cycle -- making the cache clustered and unclustering it.  (Also makes me wonder if we should have made this one easier...)

Reply:
I at least amended the text to reference what we're doing. I wonder about the 'easier' too... this makes me think a second pass at this might be wise before we cut a release.